jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -405,6 +458,44 @@ object XGBoost extends Serializable {
     xgBoostModel.setPredictionCol(predCol)
   }
 
+  private[spark] def loadPrevBooster(sparkContext: SparkContext, boosterTmpPath: String):
+      Booster = {
+    val version = getHighestVersion(sparkContext, boosterTmpPath)
+    if (version > 0) {
+      val fullPath = getTmpSavingPath(boosterTmpPath, version)
+      logger.info(s"Start training from previous booster at $fullPath")
+      val model = XGBoost.loadModelFromHadoopFile(fullPath)(sparkContext)
+      model.booster.booster.setVersion(version)
+      model.booster
+    } else {
+      null
+    }
+  }
+
+  private def getHighestVersion(sparkContext: SparkContext, boosterTmpPath: String): Int = {
+    val fs = FileSystem.get(sparkContext.hadoopConfiguration)
+    if (boosterTmpPath.isEmpty || !fs.exists(new Path(boosterTmpPath))) {
+      0
+    } else {
+      val versions = fs.listStatus(new Path(boosterTmpPath)).map(_.getPath.getName).collect {
+        case fileName if fileName.endsWith(modelSuffix) => fileName.stripSuffix(modelSuffix).toInt
+      }
+      if (versions.nonEmpty) {
+        versions.max
+      } else {
+        0
+      }
+    }
+  }
+
+  private def saveTmpBooster(sc: SparkContext, boosterTmpPath: String, model: XGBoostModel) = {
+    val fs = FileSystem.get(sc.hadoopConfiguration)
+    fs.delete(new Path(boosterTmpPath), true)
```

Comment:
sorry, missed this one, we should delete only after we successfully save the latest model, otherwise, we can lose everything

Reply:
Good point. I have made fix and renamed the method into `TmpBoosterManager.updateModel`